app.d/twitter_main.py

```diff
@@ -25,121 +25,170 @@
 
     Parameters:
         account (Account): The Twitter account object
-        campaign (Campaign): The Twitter campaign object
+        analytics (Analytics): The Twitter analytics object
         start_date (DateTime): The start date as a Deephaven DateTime object
         end_date (DateTime): The end date as a Deephaven DateTime object
         placement (str): The Twitter placement. Should be one of "ALL_ON_TWITTER" or "PUBLISHER_NETWORK"
+        entity (str): The entity of the analytics object for the API request. Should be "CAMPAIGN" or "LINE_ITEM"
     Returns:
-        (int, int, int, str): A tuple representing the clicks, engagements, and impressions on the
-            campaign for the given time period, and the full JSON response as a string.
+        str: A JSON string of the analyitcs response
     """
-    if campaign_out_of_range(campaign, start_date, end_date):
-        return (None, None, None, None)
+    if analytics_out_of_range(analytics, start_date, end_date):
+        return None
 
     metric_groups = [METRIC_GROUP.ENGAGEMENT]
     kwargs = {
         "start_time": datetime.strptime(start_date.toDateString(), "%Y-%m-%d"),
         "end_time": datetime.strptime(end_date.toDateString(), "%Y-%m-%d"),
-        "entity": "CAMPAIGN",
+        "entity": entity,
         "granularity": "HOUR",
         "placement": placement
     }
-    response = Analytics.all_stats(account, [campaign.id], metric_groups, **kwargs)
+    response = Analytics.all_stats(account, [analytics.id], metric_groups, **kwargs)
     time.sleep(4)
 
-    response_data = response[0]["id_data"][0]["metrics"]
-    clicks = response_data["clicks"]
-    engagements = response_data["engagements"]
-    impressions = response_data["impressions"]
-
-    if not (clicks is None):
-        clicks = sum(clicks)
-    if not (engagements is None):
-        engagements = sum(engagements)
-    if not (impressions is None):
-        impressions = sum(impressions)
+    return json.dumps(response)
 
-    return (clicks, engagements, impressions, json.dumps(response))
-
-def get_campaigns():
+def get_campaigns(account):
     """
     Retrieves all the campaigns for the Twitter account
 
+    Parameters:
+        account (Account): The Twitter account to pull data from
+
     Returns:
         list<Campaign>: The list of all campaigns across all accounts 
     """
     campaigns = []
-    for account in twitter_accounts:
-        for campaign in account.campaigns():
-            campaigns.append(campaign)
-        time.sleep(4)
+    for campaign in account.campaigns():
+        campaigns.append(campaign)
+    time.sleep(4)
     return campaigns
 
+def get_line_items(account):
+    """
+    Retrieves all the line items for the Twitter account
+
+    Parameters:
+        account (Account): The Twitter account to pull data from
+
+    Returns:
+        list<LineItem>: The list of all line items across all accounts 
+    """
```

Comment:
I like the documentation for this function.  Will you be doing this for the others in this file as well?

Reply:
Ha, I just got lazy with my copy and pasting and forgot to do this. I just added it, it'll be in my next PR